### PR TITLE
[INFO][EDL] Add Player.EditList to replace old Player.Cutlist

### PR DESCRIPTION
**Pull request by enen92**
## Description

The `Player.Cutlist` infolabel currently returns incorrect values:

- It includes edl cut sections which is wrong by design. The cut range is not part of the displayed timeline but rather belongs to the original file. Cuts can be added later to another GUI infolabel similar to chapters.
- It doesn't include other edit types such as EDL mutes  (only commbreaks)
- If a file has EDL cuts and other edit types such as commbreaks, the ranges control for other types will be in the wrong place in the GUI.

I've introduced `Player.Editlist` which replaces the old `Player.Cutlist`. `Player.Cutlist` will still be available to avoid breaking changes and will be removed in v21. Both infolabels return the same values. Documentation and a few improvements (getters and setters) were also added to the "interface".

@ksooo it'd be nice if you could take a look at this since `Player.Cutlist` was added by you some versions ago.

## Motivation and context
GUI consistency

## How has this been tested?
Runtime tested

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/DataCacheCore.h [hunk 1]
```diff
@@ -166,8 +163,48 @@ class CDataCacheCore
...
+      */
+    void Reset()
+    {
+      m_editList.clear();
+      m_chapters.clear();
+    }
+
+  private:
+    /*!< list of EDL edits */
+    std::vector<EDL::Edit> m_editList{};
```
Comment (ksooo) [inline xbmc/cores/DataCacheCore.h:205]: No need for the initializer list. `std::vector<EDL::Edit> m_editList;`is sufficient as deafult ctor will be called automatically.

File: xbmc/cores/DataCacheCore.h [hunk 2]
```diff
@@ -166,8 +163,48 @@ class CDataCacheCore
...
+    /*!< name and position for chapters */
+    std::vector<std::pair<std::string, int64_t>> m_chapters{};
```
Comment (ksooo) [inline xbmc/cores/DataCacheCore.h:207]: No need for the initializer list.

File: xbmc/cores/VideoPlayer/Edl.cpp [hunk 3]
```diff
@@ -733,6 +733,23 @@ int CEdl::GetTotalCutTime() const
   return m_totalCutTime; // ms
 }
 
+const std::vector<EDL::Edit> CEdl::GetEditList() const
+{
+  std::vector<EDL::Edit> editList;
+  for (EDL::Edit edit : m_vecEdits)
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/Edl.cpp:739]: You are taking a copy of the vector member and later on you modify the copy. Is this intended?

Comment (enen92) [reply xbmc/cores/VideoPlayer/Edl.cpp:739]: Yeah it is intended but I have a few doubts. I don't want to modify the original edit - hence the copy to modify later. If I was to iterate as a ref I'd risk modifying the original edit isn't it?

Comment (enen92) [reply xbmc/cores/VideoPlayer/Edl.cpp:739]: There are other places I changed that could well return const ref, will change in next revision. However here, it is intended to be a copy

Comment (ksooo) [reply xbmc/cores/VideoPlayer/Edl.cpp:739]: > However here, it is intended to be a copy
 
 That was my impression too. Just wanted to double-check with you. Maybe a comment to make clear this is by intention would be good?

Comment (enen92) [reply xbmc/cores/VideoPlayer/Edl.cpp:748]: done

**Comment by enen92** [comment]
Jenkins build this please

**Review by ksooo — APPROVED** [review]
LGTM

**Pull request merged**